cpp/src/arrow/dataset/dataset_test.cc

```diff
@@ -335,17 +332,21 @@ TEST_F(TestEndToEnd, EndToEndSingleSource) {
   // - "/2019/01 -> {"year": 2019, "month": 1}
   // - "/2019/01/CA -> {"year": 2019, "month": 1, "country": "CA"}
   // - "/2019/01/CA/a_file.json -> {"year": 2019, "month": 1, "country": "CA"}
-  auto partition_scheme = std::make_shared<SchemaPartitionScheme>(partition_schema);
-  ASSERT_OK(discovery->SetPartitionScheme(partition_scheme));
+  options.partition_scheme =
```

Comment:
Update documentation to reflect that the schema will be guessed.

Reply:
will do